### PR TITLE
variants: Add rpi_pico

### PR DESCRIPTION
**Pull request by soburi**
Add support for RaspberryPi Pico

### CODE REVIEW

**Review by DhruvaG2000 — APPROVED** [review]
I have no way of testing this ATM, but code-wise it looks good.

**Comment by DhruvaG2000** [comment]
Can you please rebase and force push?

File: variants/rpi_pico/rpi_pico.overlay [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	};
+
+	channel@2 {
+		reg = <2>;
+		zephyr,gain = "ADC_GAIN_1";
+		zephyr,reference = "ADC_REF_INTERNAL";
+		zephyr,acquisition-time = <ADC_ACQ_TIME_DEFAULT>;
+		zephyr,resolution = <12>;
+	};
+};
```
Comment (DhruvaG2000) [inline variants/rpi_pico/rpi_pico.overlay:200]: Just curious, do you want to have anything in CI for checking if this platform builds fine for atleast 1 sample?

Comment (DhruvaG2000) [reply variants/rpi_pico/rpi_pico.overlay:200]: If yes, please feel free to submit it with this PR itself

Comment (soburi) [reply variants/rpi_pico/rpi_pico.overlay:200]: > Just curious, do you want to have anything in CI for checking if this platform builds fine for atleast 1 sample?

I'm not familiar with github actions, but you might be able to combine a sample and a board in this way.

https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#about-matrix-strategies

**Pull request merged**